packages/editor/src/components/reusable-blocks-buttons/reusable-block-convert-button.js

```diff
@@ -1,24 +1,61 @@
-/**
- * External dependencies
- */
-import { every } from 'lodash';
-
 /**
  * WordPress dependencies
  */
-import { MenuItem } from '@wordpress/components';
-import { __ } from '@wordpress/i18n';
 import { hasBlockSupport, isReusableBlock } from '@wordpress/blocks';
-import { withSelect, withDispatch } from '@wordpress/data';
-import { compose } from '@wordpress/compose';
 import { BlockSettingsMenuControls } from '@wordpress/block-editor';
+import { MenuItem } from '@wordpress/components';
+import { useDispatch, useSelect } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
+
+export default function ReusableBlockConvertButton( { clientIds } ) {
+	const { isReusable, isVisible } = useSelect(
+		( select ) => {
+			const { canUser } = select( 'core' );
+			const { getBlocksByClientId, canInsertBlockType } = select(
+				'core/block-editor'
+			);
+			const { __experimentalGetReusableBlock: getReusableBlock } = select(
+				'core/editor'
+			);
+
+			const blocks = getBlocksByClientId( clientIds );
+
+			const _isReusable =
+				blocks.length === 1 &&
+				blocks[ 0 ] &&
+				isReusableBlock( blocks[ 0 ] ) &&
+				!! getReusableBlock( blocks[ 0 ].attributes.ref );
+
+			// Show 'Convert to Regular Block' when selected block is a reusable block.
+			const _isVisible =
+				isReusable ||
+				// Hide 'Add to Reusable blocks' when reusable blocks are disabled.
+				( canInsertBlockType( 'core/block' ) &&
+					blocks?.every(
```

Comment:
This is interesting while `every(undefined, something)` returns true, `undefined?.every( something )` returns undefined (aka false) which changes the meaning.

It's one of the reasons I still prefer lodash defaults over having to think about these subtile usecases. I find `something?.something` to be very problematic in a lot of usecases and something to use but not to overuse. It can quickly introduce hidden bugs everywhere.

Reply:
Lodash's `every(undefined, something)` returning `true` is a surprise to me. That's not at all what I would expect.

In this case, though, it seems like it isn't even possible for `blocks` to be undefined; so the optional chaining shouldn't even be necessary, right?

Reply:
Mathematically speaking Every(undefined) = true is the correct thing.

> In this case, though, it seems like it isn't even possible for blocks to be undefined; so the optional chaining shouldn't even be necessary, right? 

IMO, we  should assume things like that. We don't really know at this point what we have in store even if it seems technically impossible. The only way to ensure that is to have typing everywhere. 

Maybe:

 - When mounting/unmounting blocks, there's a delay where it could happen,
 - Maybe there's some async behavior (there isn't) causing this.

What I'm trying to say is that the component should be written consistently regardless of how the rest of the system works.

Reply:
In my opinion, running `every` on `undefined` isn't something one would assume to be possible by default. That sounds like just as much of a potential gotcha as the optional chaining resulting in `false` by default.

In this case, I think the current change in behavior is probably fine... if `blocks` is empty, there's nothing to convert to a reusable block at the moment, so we wouldn't want to show the option, right? In fact, that's what I had assumed the logic was already doing before my changes.

Funny enough, while discussing this, I've just found a bug in the code. We're checking for `isReusable || ...` when it should be `_isReusable || ...`. Additionally, instead of using optional chaining, we can just use nullish coalescing to default `blocks` to `[]` when it is first defined.

I'm going to go implement both of those changes real quick.